crates/bifrost/src/appender.rs

```diff
@@ -199,24 +304,29 @@ impl Appender {
                 if tone_escalated {
                     info!(
                         open_segment = %loglet.segment_index(),
+                        %error_recovery_strategy,
                         "New segment detected after {:?}",
                         total_dur
                     );
                 } else {
                     debug!(
                         open_segment = %loglet.segment_index(),
+                        %error_recovery_strategy,
                         "New segment detected after {:?}",
                         total_dur
                     );
                 }
                 return Ok(loglet);
             }
 
-            // Okay, tail segment is still sealed
+            // Okay, tail segment is still sealed or requires reconfiguration
             let log_metadata_version = log_metadata.version();
-            if start.elapsed() > auto_recovery_threshold
-                && error_recovery_strategy >= ErrorRecoveryStrategy::ExtendChainAllowed
+            if (error_recovery_strategy == ErrorRecoveryStrategy::ExtendChainPreferred
```

Comment:
Unlike in `perform_reconfiguration`, we are sleeping here between retries when using `ErrorRecoveryStrategy::ExtendChainPreferred`, right? Is this a deliberate choice between how to handle sealed/requires reconfiguration and append errors?

Reply:
I'll add a sleep, it's a good idea.